src/db/sqldb/history_sqldb_test.go

```diff
@@ -0,0 +1,347 @@
+
+			start = 0
+			end = -1
+			appId = "an-app-id"
+		})
+
+		AfterEach(func() {
+			err = hdb.Close()
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		JustBeforeEach(func() {
+			history = &models.AppScalingHistory{
+				AppId:        appId,
+				OldInstances: 2,
+				NewInstances: 4,
+				Reason:       "a reason",
+				Message:      "a message",
+			}
+
+			history.Timestamp = 666666
+			history.ScalingType = models.ScalingTypeDynamic
+			history.Status = models.ScalingStatusSucceeded
+			history.Error = ""
+			err = hdb.SaveScalingHistory(history)
+			Expect(err).NotTo(HaveOccurred())
+
+			history.Timestamp = 222222
+			history.ScalingType = models.ScalingTypeDynamic
+			history.Status = models.ScalingStatusFailed
+			history.Error = "an error"
+			err = hdb.SaveScalingHistory(history)
+			Expect(err).NotTo(HaveOccurred())
+
+			history.Timestamp = 555555
+			history.ScalingType = models.ScalingTypeSchedule
+			history.Status = models.ScalingStatusFailed
+			history.Error = "an error"
+			err = hdb.SaveScalingHistory(history)
+			Expect(err).NotTo(HaveOccurred())
+
+			history.Timestamp = 333333
+			history.ScalingType = models.ScalingTypeSchedule
+			history.Status = models.ScalingStatusIgnored
+			history.Error = ""
+			err = hdb.SaveScalingHistory(history)
+			Expect(err).NotTo(HaveOccurred())
+
+			histories, err = hdb.RetrieveScalingHistories("an-app-id", start, end)
+		})
+
+		Context("When the app has no hisotry", func() {
+			BeforeEach(func() {
+				appId = "other-app-id"
+			})
+
+			It("returns empty metrics", func() {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(histories).To(BeEmpty())
+			})
+
+		})
+
+		Context("when end time is now (end = -1)", func() {
+			BeforeEach(func() {
+				start = 333333
+				end = -1
+			})
+
+			It("returns histories from start time to now", func() {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(histories).To(HaveLen(3))
+			})
+
+		})
+
+		Context("when end time is before all the history timestamps", func() {
+			BeforeEach(func() {
+				start = 111111
+				end = 222221
+			})
+
+			It("returns empty histories", func() {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(histories).To(BeEmpty())
+			})
+
+		})
+
+		Context("when start time is after all the history timestamps", func() {
+			BeforeEach(func() {
+				start = 777777
+				end = 888888
+			})
+
+			It("returns empty histories", func() {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(histories).To(BeEmpty())
+			})
+
+		})
+
+		Context("when start time > end time", func() {
+			BeforeEach(func() {
+				start = 555555
+				end = 555533
+			})
+
+			It("returns empty histories", func() {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(histories).To(BeEmpty())
+			})
+		})
+
+		Context("when retrieving all the histories( start = 0, end = -1) ", func() {
+			It("returns all the histories of the app ordered by timestamp", func() {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(histories).To(HaveLen(4))
```

Comment:
Why not just do a Expect(histories).To(Equal([]models.AppScalingHistory{{....},{....}...}))